crates/db_views/src/post_view.rs

```diff
@@ -715,14 +712,14 @@ mod tests {
     };
     CommunityBlock::block(&conn, &community_block).unwrap();
 
-    let read_post_listings_with_person_after_block = PostQueryBuilder::create(&conn)
+    let mut read_post_listings_with_person_after_block = PostQueryBuilder::create(&conn)
       .listing_type(ListingType::Community)
       .sort(SortType::New)
-      .show_bot_accounts(false)
-      .community_id(inserted_community.id)
-      .my_person_id(inserted_person.id)
-      .list()
-      .unwrap();
+      .community_id(inserted_community.id);
+    read_post_listings_with_person_after_block.show_bot_accounts = Some(true);
+    read_post_listings_with_person_after_block.my_person_id = Some(inserted_person.id);
+    let read_post_listings_with_person_after_block =
+      read_post_listings_with_person_after_block.list().unwrap();
```

Comment:
We should add all the language stuff to these tests also. IE, have a user's preferences be All, one lang, or several langs, and make sure the queries return the correct results.

Another thing: maybe language preferences should really only count for `GetPosts`, not `Search` or `GetPost`, so that you can still search for posts in other languages, in which case we might need a flag on the PostQueryBuilder.

Reply:
The test is extremely long and confusing, i will see if i can figure that out.

`GetPost` doesnt use `PostQueryBuilder`. And the same point about `Search` could be made regarding other params like show_read_posts, which is also not supported. It would have to be a separate PR.